src/edu/ucsb/nceas/metacat/dataone/CNodeService.java

```diff
@@ -87,2025 +55,2135 @@
+     * @throws UnsupportedType
+     * @throws InsufficientResources
+     * @throws InvalidSystemMetadata
+     * @throws NotImplemented
+     * @throws InvalidRequest
+     */
+    public Identifier create(
+        Session session, Identifier pid, InputStream object, SystemMetadata sysmeta)
+        throws InvalidToken, ServiceFailure, NotAuthorized, IdentifierNotUnique, UnsupportedType,
+        InsufficientResources, InvalidSystemMetadata, NotImplemented, InvalidRequest {
+
+        try {
+            // verify the pid is valid format
+            if (!isValidIdentifier(pid)) {
+                throw new InvalidRequest("4891", "The provided identifier is invalid.");
+            }
+            if (session == null) {
+                throw new InvalidToken("4894", "Session is required to WRITE to the Node.");
+            }
+            logMetacat.debug("CN.create -start to create the object with pid " + pid.getValue());
+            // The lock to be used for this identifier
+            //Lock lock = null;
+
+            try {
+
+
+                // are we allowed?
+                boolean isAllowed = false;
+                D1AuthHelper authDel = new D1AuthHelper(request, pid, "4861", "????");
+                authDel.doCNOnlyAuthorization(session);
+
+                isAllowed = true;
+                // proceed if we're called by a CN
+                if (isAllowed) {
+                    objectExists(pid);
+
+                    //check if the series id is legitimate. It uses the same rules of the method
+                    // registerSystemMetadata
+                    //checkSidInModifyingSystemMetadata(sysmeta, "4896", "4893");
+                    Identifier sid = sysmeta.getSeriesId();
+                    if (sid != null) {
+                        if (!isValidIdentifier(sid)) {
+                            throw new InvalidRequest(
+                                "4891",
+                                "The series id in the system metadata is invalid in the request.");
+                        }
+                    }
+                    // create the coordinating node version of the document
+                    logMetacat.debug(
+                        "CN.create - after locking identifier, passing authorization check, "
+                            + "continue to create the object "
+                            + pid.getValue());
+                    sysmeta.setSerialVersion(BigInteger.ONE);
+                    //for the object whose authoritative mn is v1. we need reset the modification
+                    // date.
+                    //for the object whose authoritative mn is v2. we just accept the
+                    // modification date.
+                    D1NodeVersionChecker checker =
+                        new D1NodeVersionChecker(sysmeta.getAuthoritativeMemberNode());
+                    String version = checker.getVersion("MNRead");
+                    boolean changeModificationDate =
```

Comment:
Did you make the change or the editor make?

Reply:
disregard - will revert changes and recommit without code cleanup